### PR TITLE
School Library UI

### PR DESCRIPTION
**Pull request by Javivy**
## In this milestone I implemented the following functions and enhancements:

  - [x] Created `App` class
  - [x] Created several methods

1. `list_all_books`: Prints a list of all the books created, its title, and author
2. `list_all_people`: Prints a list of all the people created if it's a Student or a Teacher, its name, ID, and age
3. `create_person`: Allows the user to create a person (Student or teacher), waiting for the user to input data
4. `create_book`: Allows the user to create a book, waiting for the user to input the data (Title and Author)
5. `create_rental`: Allows the user to create a rental for a book and a person, waiting for the user to input data (ID)
6. `list_rentals_by_person_id`: Prints a list of all the rentals created for a given ID 

  - [x] Created `main.rb` to run the application

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+      parent_permission_console = gets.chomp.capitalize
+      if parent_permission_console == 'N'
+        @people.push(Student.new(age, name: name, parent_permission: false))
+      elsif parent_permission_console == 'Y'
+        @people.push(Student.new(age, name: name, parent_permission: true))
+      else
+        puts 'Please select between [Y/N]'
+      end
+      puts('Person created successfully')
+    when 2
```
Comment (Reem-lab) [inline app.rb:89]: - [ ] Great job working on the functionality so far but when I try to create a new student, I face this error with the wrong number of arguments (given 2, expected 1), so kindly check on this error and try to solve the problem to let the user be able to create a student ⭐🔥💯🙏🏻

- When I try to create a student  ⬇
image

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+      print('Specialization: ')
+      specialization = gets.chomp
+      print('Name: ')
+      name = gets.chomp
+      @people.push(Teacher.new(age, specialization, name: name))
+      puts('Person created successfully')
+    else
+      puts("The number #{number} isn't a valid choice, please select number (1) to create a student or (2) to create
+      a teacher")
+    end
```
Comment (Reem-lab) [inline app.rb:101]: - [ ] Great job working on the functionality so far but when I try to create a new teacher, I face this error with the wrong number of arguments (given 3, expected 1), so kindly check on this error and try to solve the problem to let the user be able to create a teacher⭐🔥💯🙏🏻

- When I try to create a student teacher⬇
image

**Review by Reem-lab — CHANGES_REQUESTED** [review]
# Hi @Javivy, 👋🏻👋🏻

Great job so far! ✨⭐🤩✅
There are some issues that you still need to work on to go to the next project but you are almost there!

### Note to the next reviewer: Kindly check the rentals since I couldn't create a new person so I couldn't be able to check on the `create rental` and `list rental`  methods✅🤩



## Highlights
- [ ] No Linters error ✅
- [ ] Followed Gitflow ⭐
- [ ] Great job working on the whole functionality 💯✨
- [ ] Descriptive PR 🚀

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Felix45 — APPROVED** [review]
# Approved :green_circle: 
Hi @Javivy, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights :heavy_check_mark: 
- [x] Good UI design :white_check_mark: 
- [x] Git flow is used correctly :white_check_mark: 
- [x] Pull request has a descriptive title :white_check_mark:   
- [x] Implemented all the changes from the previous reviewer :white_check_mark:  

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**